src/planner/binder/statement/bind_update.cpp

```diff
@@ -202,10 +212,59 @@ BoundStatement Binder::Bind(UpdateStatement &stmt) {
 	// set the projection as child of the update node and finalize the result
 	update->AddChild(move(proj));
 
-	result.names = {"Count"};
-	result.types = {LogicalType::BIGINT};
-	result.plan = move(update);
-	this->allow_stream_result = false;
+	if (!stmt.returning_list.empty()) {
+		auto types = vector<LogicalType>();
+		auto names = vector<std::string>();
+
+		auto update_table_index = GenerateTableIndex();
+		update->table_index = update_table_index;
+
+		auto binder = Binder::CreateBinder(context);
+
+		for (auto &col : table->columns) {
+			names.push_back(col.name);
+			types.push_back(col.type);
+		}
+
+		binder->bind_context.AddGenericBinding(update_table_index, table->name, names, types);
```

Comment:
Most of the code here seems shared with the `bind_insert`, could we extract this into a function and re-use it in those two places?